### ISSUE TITLE
'NoneType' object has no attribute 'rollback'

### ISSUE THREAD

**Issue by degtyaryov** [2ndquadrant-it/barman#252]
Hello.

My Barman doing backups for 5 db.

I doing backups from slaves.
It so happened that on one of the masters the blocks were damaged.
I made a slave master and re-created the slave. Swapped them.

Now I can not backup.

Barman:
```
ERROR: Backup failed issuing start backup command. DETAILS: Connection lost, reconnection not allowed
```

Postgres Slave:
```
LOG:  could not receive data from client: Connection timed out
WARNING:  aborting backup due to backend exiting before pg_stop_backup was called
```

I saw the issue https://github.com/2ndquadrant-it/barman/issues/149.
I update barman from 2.8 to 2.9.

Barman:
```
ERROR: Backup failed issuing start backup command. DETAILS: 'NoneType' object has no attribute 'rollback'
```

Postgres Slave:
```
LOG:  could not receive data from client: Connection timed out
WARNING:  aborting backup due to backend exiting before pg_stop_backup was called
```

Everything is OK with the Network!

DB Size 1.3 Tb.

Other 4 db(> 2 Tb) backups are done successfully.

What is the problem?

Thanks!

**Comment by gbartolini** [comment]
I fear we might need more information to debug this issue.

**Comment by mnencia** [comment]
Duplicated of  #239 
A fix has already been commited and will be released with Barman 2.10
